monitoring/src/main/java/org/polypheny/db/monitoring/events/analyzer/DmlEventAnalyzer.java

```
@@ -0,0 +1,96 @@
+                .executionTime( dmlEvent.getExecutionTime() )
+                .rowCount( dmlEvent.getRowCount() )
+                .isSubQuery( dmlEvent.isSubQuery() )
+                .recordedTimestamp( dmlEvent.getRecordedTimestamp() )
+                .build();
+
+        RelRoot relRoot = dmlEvent.getRouted();
+        if ( relRoot != null ) {
+            RelNode node = relRoot.rel;
+            processRelNode( node, dmlEvent, metric );
+        }
+
+        if ( dmlEvent.isAnalyze() ) {
+            processDurationInfo( dmlEvent, metric );
+        }
+
+        return metric;
+    }
+
+
+    private static void processDurationInfo( DmlEvent dmlEvent, DmlDataPoint metric ) {
+        try {
+            InformationDuration duration = new Gson().fromJson( dmlEvent.getDurations(), InformationDuration.class );
+            getDurationInfo( metric, "Plan Caching", duration );
+            getDurationInfo( metric, "Index Lookup Rewrite", duration );
+            getDurationInfo( metric, "Constraint Enforcement", duration );
+            getDurationInfo( metric, "Implementation Caching", duration );
+            getDurationInfo( metric, "Index Update", duration );
+            getDurationInfo( metric, "Routing", duration );
+            getDurationInfo( metric, "Planning & Optimization", duration );
+            getDurationInfo( metric, "Implementation", duration );
+            getDurationInfo( metric, "Locking", duration );
+        } catch ( Exception e ) {
+            log.debug( "could not deserialize of get duration info" );
+        }
+    }
+
+
+    private static void getDurationInfo( DmlDataPoint dmlMetric, String durationName, InformationDuration duration ) {
+        try {
+            long time = duration.getNanoDuration( durationName );
+            dmlMetric.getDataElements().put( durationName, time );
+        } catch ( Exception e ) {
+
+            log.debug( "could no find duration:{}", durationName );
```

This hides an error...